mava/systems/jax/config_test.py

```diff
@@ -0,0 +1,252 @@
+    param_2: str
+
+
+@pytest.fixture
+def dummy_component_config() -> ComponentConfig:
+    """Dummy config dataclass for a component.
+
+    Returns:
+        config dataclass
+    """
+    return ComponentConfig(name="component", setting=5)
+
+
+@pytest.fixture
+def dummy_hyperparameter_config() -> HyperparameterConfig:
+    """Dummy config dataclass for component hyperparameters.
+
+    Returns:
+        config dataclass
+    """
+    return HyperparameterConfig(param_0=2.7, param_1=3.8)
+
+
+@pytest.fixture
+def config() -> Config:
+    """Config instance.
+
+    Returns:
+        instantiation of a Mava Config class
+    """
+    return Config()
+
+
+def test_add_single_config(config: Config, dummy_component_config: type) -> None:
+    """Test adding a single config.
+
+    Args:
+        config : Mava config
+        dummy_component_config : component config dataclass
+    """
+    config.add(component=dummy_component_config)
+    config.build()
+    conf = config.get()
+
+    assert conf.name == "component"
+    assert conf.setting == 5
+
+
+def test_add_multiple_configs(
+    config: Config, dummy_component_config: type, dummy_hyperparameter_config: type
+) -> None:
+    """Test adding multiple configs at the same time.
+
+    Args:
+        config : Mava config
+        dummy_component_config : component config dataclass
+        dummy_hyperparameter_config : component config dataclass of hyperparameters
+    """
+    config.add(
+        component=dummy_component_config, hyperparameter=dummy_hyperparameter_config
+    )
+    config.build()
+    conf = config.get()
+
+    assert conf.name == "component"
+    assert conf.setting == 5
+    assert conf.param_0 == 2.7
+    assert conf.param_1 == 3.8
+
+
+def test_add_configs_twice(
+    config: Config, dummy_component_config: type, dummy_hyperparameter_config: type
+) -> None:
+    """Test add two configs, one after the other.
+
+    Args:
+        config : Mava config
+        dummy_component_config : component config dataclass
+        dummy_hyperparameter_config : component config dataclass of hyperparameters
+    """
+    config.add(component=dummy_component_config)
+    config.add(hyperparameter=dummy_hyperparameter_config)
+    config.build()
+    conf = config.get()
+
+    assert conf.name == "component"
+    assert conf.setting == 5
+    assert conf.param_0 == 2.7
+    assert conf.param_1 == 3.8
+
+
+def test_update_config(
+    config: Config, dummy_component_config: type, dummy_hyperparameter_config: type
+) -> None:
+    """Test add two configs, one after the other.
+
+    Args:
+        config : Mava config
+        dummy_component_config : component config dataclass
+        dummy_hyperparameter_config : component config dataclass of hyperparameters
+    """
+    config.add(component=dummy_component_config)
+    config.update(component=dummy_hyperparameter_config)
+    config.build()
+    conf = config.get()
+
+    assert conf.param_0 == 2.7
+    assert conf.param_1 == 3.8
+
```

Comment:
Possibly add `assert not hasattr(config,"setting")` since we shouldn't have this prop anymore.

Reply:
Good idea. Will do

Reply:
Done